File: FacilitiesManagementAPI/Entities/Contractor.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace FacilitiesManagementAPI.Entities
{
    public class Contractor
    {
        public Guid Id { get; set; }
        public string BusinessName { get; set;  }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int? ContractorTypeId { get; set; }
        public int GreenLightId { get; set; }
        public string PhoneNumber1 { get; set; }
        public string PhoneNumber2 { get; set; }
        public string Email { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime DateCreated { get; set; } = DateTime.Now;
        public ICollection<Premises> Premises {  get; set; }
        public ICollection<Note> Notes { get; set; }
        public ICollection<PremisesTask> Jobs { get; set; }

    }
}

File: client/src/app/_models/contractor.ts
import { Note } from "./note";
import { PremisesTask } from "./premisesTask";

export interface Contractor {
    id: string;
    businessName: string;
    firstName: string;
    lastName: string;
    rating: number;
    contractorTypeId: number;
    greenLightEnum: string;
    phoneNumber1: string;
    phoneNumber2: string;
    email: string;
    isDeleted: boolean;
    dateCreated: Date;
    notes: Note[];
    jobs: PremisesTask[];
}